blinker/base.py

```diff
@@ -440,7 +443,9 @@ class WeakNamespace(WeakValueDictionary):
 
     """
 
-    def signal(self, name, doc=None):
+    signal_cls = NamedSignal
+
+    def signal(self, name, doc=None, signal_cls=None):
```

Comment:
I'm very wary of this. With this it is trivial to break the contract that multiple calls to `signal()` return the same object. Two callers, one supplies `signal_cls`, one does not. Depending on the call sites it may not even be deterministic which one wins. Regardless, one caller gets what they expect and one doesn't.